app/controllers/endpoints_controller.rb

```diff
@@ -39,8 +39,15 @@ def update
   end
 
   def test
-    SendNotificationJob.perform_now(@endpoint, Time.zone.now, 'Test', 'Test message from webhooks')
-    head :no_content
+    job = SendNotificationJob.perform_now(@endpoint, Time.zone.now, 'Test', 'Test message from webhooks')
+    if job.nil?
+      head :no_content
+      return
+    end
+
+    status = :bad_gateway
```

Comment:
I think we should catch an exception here. Otherwise, if there is an exception, it will never come here.

Reply:
Last time I checked `job` is `nil` if the job execution succeeded, otherwise ActiveJob catches the exception and returns it.

Reply:
Not if the endpoint is not ready. Got this log from running only rails app:

```
[ActiveJob] [SendNotificationJob] [52763967-bfba-40db-997d-e9489f0db0f3] Error performing SendNotificationJob (Job ID: 52763967-bfba-40db-997d-e9489f0db0f3) from Resque(default) in 4353.9ms: SocketError (Failed to open TCP connection to endpoint:4567 (getaddrinfo: Name or service not known)):
...
snip callstack
...
Completed 500 Internal Server Error in 4438ms (Views: 0.7ms | ActiveRecord: 15.5ms)
```